### PR TITLE
feat: copy button on snippet code

### PR DESCRIPTION
**Pull request by ramaID**

### CODE REVIEW

**Comment by satnaing** [comment]
Seems pretty useful.
How about we place the button inside the code block; and adjust the border-radius to be less rounded? 
I think it looks nicer.

Screenshot 2024-01-07 at 6 37 32 PM

File: src/layouts/PostDetails.astro [hunk 1]
```diff
@@ -106,4 +106,53 @@ const layoutProps = {
...
+      let wrapper = document.createElement("div");
+      wrapper.style.position = "relative";
+
+      let copyButton = document.createElement("button");
+      copyButton.className = "copy-code";
+      copyButton.innerHTML = copyButtonLabel;
+      copyButton.style.position = "absolute";
+      copyButton.style.top = '-10px';
+      copyButton.style.right = '10px';
+      copyButton.style.backgroundColor = "#3730a3";
```
Comment (satnaing) [inline src/layouts/PostDetails.astro:124]: Also can you make the background color of the button to be a part of the theme color?

Comment (satnaing) [reply src/layouts/PostDetails.astro:124]: I can co-author this if necessary.

Comment (ramaID) [reply src/layouts/PostDetails.astro:124]: Yes, please. I have no experience with UI/UX, because I work mostly on the backend.

**Comment by satnaing** [comment]
Just make some changes and merged it.
Thanks for your PR @ramaID.

**Pull request merged**